### PR TITLE
docs: add filedump docs

### PR DESCRIPTION
**Pull request by TomShawn**
docs: add filedump docs

ref: https://github.com/cloudberrydb/filedump

---

## Change logs

> Describe your change clearly, including what problem is being solved or what document is being added or updated.

## Contributor's checklist

Here are some reminders before you submit your pull request:

* Make sure that your Pull Request has a clear title and commit message. You can take the [Git commit template](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) as a reference.
* Sign the Contributor License Agreement as prompted for your first-time contribution (*One-time setup*).
* Learn the [code contribution](https://cloudberrydb.org/contribute/code) and [doc contribution](https://cloudberrydb.org/contribute/doc) guides for better collaboration.
* Make sure that your changes deployment preview is successful.
* List your communications in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb-site/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
* Feel free to ask for the @cloudberrydb/doc team or other people to help review and approve.

### CODE REVIEW

File: docs/sys-utilities/pg-filedump.md [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+- **File analysis and debugging**: Converts heap, index, or control files from the database into a readable format, helping file content analysis and database issue debugging.
+- **Data recovery**: Helps extract and recover data when database files are corrupted.
+- **Database learning and research**: Provides developers and administrators a way to directly inspect database file contents, which is useful for studying database internals or learning about file formats.
+
+
+## Compile and install filedump
+
+Before compiling filedump, you need to have a Cloudberry Database cluster installed with version 1.0.0 or later. The following are the typical steps for compilation:
+
+1. Ensure the `gpdb` package can be found:
```
Comment (tuhaihe) [inline docs/sys-utilities/pg-filedump.md:24]: gpdb ~> Cloudberry ?

File: i18n/zh/docusaurus-plugin-content-docs/current/sys-utilities/pg-filedump.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+- 文件分析与调试：将数据库的堆文件、索引文件或控制文件转化为可读格式，有助于分析文件内容、调试数据库问题。
+- 数据恢复：在数据库文件损坏时，使用 filedump 可以帮助提取数据并进行恢复。
+- 数据库学习与研究：对于研究数据库内部结构或学习数据库文件格式的开发者和管理员，filedump 提供了直接查看数据库文件内容的手段。
+
+## 编译与安装
+
+在编译 filedump 前，你需要安装好一套 Cloudberry Database v1.0.0 及以上版本的集群，以下是典型的编译步骤：
+
+1. 确保 gpdb 包可以被找到：
```
Comment (tuhaihe) [inline i18n/zh/docusaurus-plugin-content-docs/current/sys-utilities/pg-filedump.md:23]: gpdb ~> Cloudberry ?

**Comment by tuhaihe** [comment]
Thanks for your great work! @TomShawn 
Only two comments. FYI.

**Review by tuhaihe — APPROVED** [review]

**Review by EcaleD — APPROVED** [review]
LGTM.

**Pull request merged**